### PR TITLE
docs(component-testing): Added rerender/updating props recipe

### PR DESCRIPTION
**Pull request by agg23**
### User facing changelog
Provided recipe example of how to update props without remounting in component testing.

### How has the user experience changed?
Documentation update only; no user experience changes.

### CODE REVIEW

File: npm/react/docs/recipes.md [hunk 1]
```diff
@@ -158,3 +164,83 @@ replace({ 'process.env.NODE_ENV': JSON.stringify('development') }),
...
+      ))}
+    </ul>
+  )
+}
+```
+
+This component is an accumulator that stores each `value` prop passed to it. We create a wrapper component that has an `input` and a `button` to push new values to the `value` prop.
+
+```js
+const Wrapper = () => {
```
Comment (dmtrKovalenko) [inline npm/react/docs/recipes.md:203]:
```suggestion
const TestAcc = () => {
```

**Comment by elevatebart** [comment]
The new recipe is very useful as is.
Could you add a mention of rerender somewhere please?
This is what users are going to be searching for.
Thet's give them a leg up ;)

File: npm/react/docs/recipes.md [hunk 2]
```diff
@@ -158,3 +164,83 @@ replace({ 'process.env.NODE_ENV': JSON.stringify('development') }),
...
 
 See [examples/rollup](examples/rollup) folder for full example.
+
+## Usage
+
+### Changing props
+
+Many components have some statefulness, whether explicitly through `useState`, or implicitly through `useEffect`. Therefore during testing it is useful to keep the component mounted, but change the props being passed to it in order to preserve its state.
+
+We recommend building a "wrapper" component with the necessary DOM controls to push new props to your component under test.
```
Comment (dmtrKovalenko) [inline npm/react/docs/recipes.md:174]:
```suggestion
We recommend building a "wrapper" component, that will be close to how your users will use this component. And add DOM controls that will change props during the test. 
```

**Review by elevatebart — CHANGES_REQUESTED** [review]
The new recipe is very useful as is.
Could you add a mention of rerender somewhere please?
This is what users are going to be searching for.
Let's give them a leg up ;)

**Review by lmiller1990 — APPROVED** [review]

**Comment by lmiller1990** [comment]
Need approval from @elevatebart to merge.

**Review by elevatebart — APPROVED** [review]
Hope to get the rerender function in there soon... people will ask for it.

**Pull request merged**